### PR TITLE
Task/simplify build scripts

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR simplifies the build scripts we maintain to test compilers on LC systems and makes them much easier to maintain.
- Specifically, there is now one script for each compiler family (e.g., clang, gcc, nvcc+xl) and you pass the compiler version(s) you want to build with to each script.
- Hopefully, now we can just run them with new compiler versions and we won't have to modify/add scripts.

### CODE REVIEW

**Review by artv3 — APPROVED** [review]

**Review by davidbeckingsale — APPROVED** [review]
I like this. Do you think we should have the scripts pick some default version when no argument is passed?

**Comment by rhornung67** [comment]
> I like this. Do you think we should have the scripts pick some default version when no argument is passed?

I thought about that. We could do that easily enough. Default is what module is loaded?

Not sure what people want or who uses these. I use them all the time to test compilers and it has become a maintenance burden.

**Comment by rhornung67** [comment]
@MrBurmark @mdavis36 @rchen20 you folks use these build scripts, correct? Are you OK with these changes? It makes them much easier to maintain.

**Review by MrBurmark — APPROVED** [review]

File: scripts/lc-builds/blueos_nvcc_clang.sh [hunk 1]
```diff
@@ -16,12 +31,13 @@ module load cmake/3.14.5
 
 cmake \
   -DCMAKE_BUILD_TYPE=Release \
-  -DCMAKE_CXX_COMPILER=/usr/tce/packages/clang/clang-upstream-2019.08.15/bin/clang++ \
+  -DCMAKE_CXX_COMPILER=/usr/tce/packages/clang/clang-${COMP_CLANG_VER}/bin/clang++ \
+  -DBLT_CXX_STD=c++11 \
```
Comment (rchen20) [inline scripts/lc-builds/blueos_nvcc_clang.sh:35]: This is BLT_CXX_STD is a long awaited good addition. In one of my branches I added this in the top level CMakeLists.txt, not sure if you want to move it there?

**Review by rchen20 — APPROVED** [review]
I like this change, it should improve maintenance of our build scripts drastically. Just had one comment about the BLT_CXX_STD flag being moved to the top level CMakeLists.txt; everything else looks good.

File: scripts/lc-builds/blueos_nvcc_clang.sh (hunk 1 above)
Comment (rhornung67) [reply scripts/lc-builds/blueos_nvcc_clang.sh:35]: I don't think we want to force it since BLT doesn't and users should be aware of it.

**Comment by mdavis36** [comment]
> @MrBurmark @mdavis36 @rchen20 you folks use these build scripts, correct? Are you OK with these changes? It makes them much easier to maintain.

@rhornung67 I use these all the time, I like the changes. +1 to maybe having some defaults defined for each of the scripts, perhaps each script could be set with a compiler version we use to test our gitlab CI? I think it would also be nice to have a check to ensure the compilers we request actually exist on the system before creating a new directory so we don't accidentally create useless directories whenever there is a typo

**Comment by rhornung67** [comment]
@mdavis36 agree with most of what you wrote. We can spice these up in a subsequent PR. Personally, I don't want to insert compiler version stuff to reintroduce maintenance burden. Perhaps, if no version is specified use the one corresponding to the module that is currently loaded?

**Pull request merged**